src/pages/Environments.js

```diff
@@ -134,9 +140,14 @@ const Environments = () => {
       Ajax(signal).Disks.list({ creator, includeLabels: 'saturnApplication,saturnWorkspaceNamespace,saturnWorkspaceName' }),
       Ajax(signal).Apps.listWithoutProject({ creator, includeLabels: 'saturnWorkspaceNamespace,saturnWorkspaceName' })
     ])
-    setRuntimes(newRuntimes)
-    setDisks(newDisks)
-    setApps(newApps)
+
+    // Old apps, runtimes and disks may not have 'saturnWorkspaceNamespace' label defined. When they were
+    // created, workspace namespace (a.k.a billing project) value used to equal the google project.
+    // Therefore we use google project if the namespace label is not defined.
+    setRuntimes(_.map(addNamespaceLabelIfAbsent, newRuntimes))
+    setDisks(_.map(addNamespaceLabelIfAbsent, newDisks))
+    setApps(_.map(addNamespaceLabelIfAbsent, newApps))
```

Comment:
This is the entry point that felt right to handle any missing workspace namespace labels at, to contain the logic in a single place.

Reply:
If it's only used here, should this be `addNamespaceLabelsIfAbsent`, with the map inside the function?